config/confighttp/compression.go

```diff
@@ -146,6 +147,7 @@ func (d *decompressor) wrap(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		newBody, err := newBodyReader(r)
 		if err != nil {
+			r.Body.Close()
 			d.errorHandler(w, r, err.Error(), http.StatusBadRequest)
 			return
 		}
```

Comment:
Is there not a bug at line 155? Why closing the newBody?

Reply:
Please address this comment?